### PR TITLE
imp: transfer total escrow follow ups

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

feature branch containing changes approved in smaller prs:
- #3509 
- #3519
- #3506 
- #3605

ref: #3368 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testvalues/values.go
Comment (damiannolan) [inline e2e/testvalues/values.go:1]: One more to add here after conflicts are resolved https://github.com/cosmos/ibc-go/pull/3553#discussion_r1203972599

Comment (damiannolan) [inline e2e/testvalues/values.go:1]: we could use a var block to declare all feature releases but I think having them separate is also nice and readable!

**Review by damiannolan — APPROVED** [review]
Everything in this PR looks good to me! Squeaky clean 🧽

**Review by colin-axner — COMMENTED** [review]
ACK, LGTM

**Review by damiannolan — APPROVED** [review]
ayy! ACK latest commits since review. LGTM 🚀

**Review by DimitrisJim — APPROVED** [review]

**Pull request merged**